Aligning ProwlerMemberRole inline policy with latest prowler version check access requirements

vfeinman opened the pull request:
*Description of changes:*
Update Prowler IAM permissions in member role template

Enhanced the IAM permissions in the 1-sat2-member-roles.yaml template to align with actions listed in upstream prowler repo https://github.com/prowler-cloud/prowler/blob/master/permissions/prowler-additions-policy.json and from prowler error output:
- Added backup:Get* permissions
- Added bedrock:List* and bedrock:Get* permissions
- Added codebuild:ListReportGroups permission
- Added glue:GetMLTransforms permission
- Added macie2:GetAutomatedDiscoveryConfiguration permission
- Added servicecatalog:Describe* and servicecatalog:List* permissions
- Added states:ListTagsForResource permission
- Added ssm:DescribeDocumentPermission permissions



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

vfeinman committed:
Update Prowler IAM permissions in member role template

Enhanced the IAM permissions in the 1-sat2-member-roles.yaml template to align with actions listed in upstream prowler repo https://github.com/prowler-cloud/prowler/blob/master/permissions/prowler-additions-policy.json:
- Added backup:Get* permissions
- Added bedrock:List* and bedrock:Get* permissions
- Added codebuild:ListReportGroups permission
- Added glue:GetMLTransforms permission
- Added macie2:GetAutomatedDiscoveryConfiguration permission
- Added servicecatalog:Describe* and servicecatalog:List* permissions
- Added states:ListTagsForResource permission

vfeinman committed:
Update Prowler IAM permissions in member role template

Enhanced the IAM permissions in the 1-sat2-member-roles.yaml template based on error output from prowler assessment:
- Added ssm:DescribeDocumentPermission permissions

js37 commented:
These same changes will also need to be added to [2-sat2-codebuild-prowler.yaml](https://github.com/awslabs/aws-security-assessment-solution/blob/b8a144624cb1f0d41f923ee1d545e3a190c48c0d/2-sat2-codebuild-prowler.yaml#L181)

vfeinman committed:
Update Prowler IAM permissions in member role template

Enhanced the IAM permissions in the 1-sat2-member-roles.yaml template to align with actions listed in upstream prowler repo https://github.com/prowler-cloud/prowler/blob/master/permissions/prowler-additions-policy.json and from prowler error output:

    Added backup:Get* permissions
    Added bedrock:List* and bedrock:Get* permissions
    Added codebuild:ListReportGroups permission
    Added glue:GetMLTransforms permission
    Added macie2:GetAutomatedDiscoveryConfiguration permission
    Added servicecatalog:Describe* and servicecatalog:List* permissions
    Added states:ListTagsForResource permission
    Added ssm:DescribeDocumentPermission permissions

vfeinman commented:
@js37 Added same changes to [2-sat2-codebuild-prowler.yaml](https://github.com/awslabs/aws-security-assessment-solution/pull/25/commits/d0d407fffeb99e42854a0e2cafa3ce6d6d112177#diff-e41d6b1f81e3e5353c9fcc1455a6a03069290eb625cd2f5e519ddee27861b2d2).
